### PR TITLE
Actually port plonetheme.teamraum tests to opengever.core testing

### LINKED ISSUE #4725: Test isolation failure in plonetheme.teamraum functional tests
Quick reproduction:

```
$ bin/test -m plonetheme.teamraum
--- snip ---
Total: 23 tests, 0 failures, 2 errors, 0 skipped in 35.040 seconds.
```

### PR DESCRIPTION
**Pull request by Rotonen**
Reduces the unnecessary background noise in our tests by a lot.

Closes #4725

### CODE REVIEW

File: plonetheme/teamraum/testing.py [hunk 1]
```diff
@@ -1,47 +1,42 @@
...
-        login(portal, TEST_USER_NAME)
-        # Install into Plone site using portal_setup
         applyProfile(portal, 'plonetheme.teamraum:default')
 
-    def tearDown(self):
-        super(TeamraumThemeLayer, self).tearDown()
-        clear_transmogrifier_registry()
 
+PLONETHEME_TEAMRAUM_FIXTURE = TeamraumThemeFunctionalLayer()
+PLONETHEME_TEAMRAUM_FUNCTIONAL_TESTING = FunctionalTesting(
```
Comment (phgross) [inline plonetheme/teamraum/testing.py]: I do not see any need, of a specific PLONETHEME_TEAMRAUM layer, just uses GEVER's IntegrationTestCase

Comment (Rotonen) [reply plonetheme/teamraum/testing.py]: As discussed with @lukasgraf, we should apply the theme profile onto our normal tests.

For now I'd say this is fine if we later squash the layers into one and address the transform issues / our testing to figure out which is wrong.

File: plonetheme/teamraum/tests/test_searchbox.py [hunk 2]
```diff
@@ -2,28 +2,17 @@
...
 
-class TestSeachBoxViewlet(TestCase):
-
-    layer = TEAMRAUMTHEME_FUNCTIONAL_TESTING
-
-    def setUp(self):
-        self.portal = self.layer['portal']
-        self.request = self.layer['request']
-        setRoles(self.portal, TEST_USER_ID, ['Manager'])
+class TestSeachBoxViewlet(TeamraumThemeTestCase):
```
[the remark this answers is not in the document]
Comment (Rotonen) [reply plonetheme/teamraum/tests/test_searchbox.py]: I'll drop the pre-existing testing duplication here.

File: plonetheme/teamraum/tests/test_personal_bar.py [hunk 3]
```diff
@@ -1,21 +1,10 @@
...
 
-class TestSearchBoxViewlet(TestCase):
-
-    layer = TEAMRAUMTHEME_FUNCTIONAL_TESTING
-
-    def setUp(self):
-        self.portal = self.layer['portal']
-        self.request = self.layer['request']
-        setRoles(self.portal, TEST_USER_ID, ['Manager'])
+class TestSearchBoxViewlet(TeamraumThemeTestCase):
```
[the remark this answers is not in the document]
Comment (Rotonen) [reply plonetheme/teamraum/tests/test_personal_bar.py]: Good catch, will verify.

[the remark this answers is not in the document]
Comment (Rotonen) [reply plonetheme/teamraum/tests/test_personal_bar.py]: This one is actually just a bad class name as this actually tests the personal bar language selector and not at all the search box. Probably an old copy paste fail.

**Comment by Rotonen** [comment]
@phgross countersuggestion:

Split off the searchbox deduplication hunt / verification and the profile application / layer squash onto its own issue.

**Comment by Rotonen** [comment]
@phgross created the split issue at https://github.com/4teamwork/opengever.core/issues/5008

For the sake of formality, approve and merge?

**Pull request merged**